lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -2138,24 +2138,24 @@ namespace Js
         public:
             AutoRestoreFunctionInfo(ParseableFunctionInfo *pfi) : pfi(pfi), funcBody(nullptr) {}
             ~AutoRestoreFunctionInfo() {
-                if (this->funcBody && this->funcBody->GetFunctionInfo()->GetFunctionProxy() == this->funcBody)
+                if (this->pfi != nullptr)
                 {
-                    FunctionInfo *functionInfo = funcBody->GetFunctionInfo();
+                    FunctionInfo *functionInfo = this->pfi->functionInfo;
```

Comment:
Knowing the above, would the following suffice:

if (this->pfi != nullptr && this->pfi->GetFunctionInfo()->GetFunctionProxy() == this->pfi)